node/core/candidate-validation/src/lib.rs

```diff
@@ -500,7 +558,7 @@ fn perform_basic_checks(
 		return Err(InvalidCandidate::PoVHashMismatch)
 	}
 
-	if *validation_code_hash != candidate.validation_code_hash {
+	if let Some(false) = validation_code_hash.map(|hash| *hash == candidate.validation_code_hash) {
```

Comment:
In fact it is necessary to preserve it, candidate validation subsystem still can receive `ValidateFromExhaustive` request.

Reply:
Yeah, what I am saying above is basically: it seems fine this way. The rationale is:
- if the validation code was provided via the exhaustive validation request, then we do have to perform this check. The `ValidationCodeHash` will be Some and we do the check.
- if we are on the validation from onchain state path, then the code hash won't be provided but that's fine because the only way to get it via querying code by hash which implicitly preserves this invariant.